### PR TITLE
Fix AstTypeStubPaths* tests

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
```cs
                analysis.Should().HavePythonModuleVariable("Package")
                    .Which.Should().HaveNestedModule("Module")
                    .Which.Should().HaveMultipleTypesMember("Class");
```
does not always apply, the structure is different in all cases. Original tests only verified actual variable members for completion.

### CODE REVIEW

**Comment by MikhailArkhipov** [comment]
WRT  `FindExpressionsForDefinition` test - it was failing that expectation for def location on `.` position was `no expression`. IMO it is OK if instead it finds `C().f(1)` since `.` is the root operator here.

File: src/Analysis/Engine/Test/AstAnalysisTests.cs [hunk 1]
```diff
@@ -381,91 +377,91 @@ class BankAccount(object):
...
                     .WithValue<IBuiltinInstanceInfo>()
-                    .Which.Should().HaveMemberType(PythonMemberType.Instance)
-                    .And.HavePythonType(type)
-                    .Which.Should().HaveMethod("typed_method_2")
+                    .Which.Should().HaveMembers("typed_method_2")
                     .And.NotHaveMembers("untyped_method", "inferred_method", "typed_method");
             }
         }
 
+        private async Task<IModuleAnalysis> GetStubBasedAnalysis(
```
Comment (AlexanderSher) [inline src/Analysis/Engine/Test/AstAnalysisTests.cs]: It doesn't look like this method has reduced the amount of code.

Comment (MikhailArkhipov) [reply src/Analysis/Engine/Test/AstAnalysisTests.cs]: It is more if we want to add additional tests with actual multiple paths and stubs. Existing one are rather primitive.

File: src/Analysis/Engine/Test/FluentAssertions/AnalysisValueAssertions.cs [hunk 2]
```diff
@@ -95,6 +95,22 @@ public AndConstraint<TAssertions> HaveMembers(params string[] memberNames)
             return new AndConstraint<TAssertions>((TAssertions)this);
         }
 
+        public AndConstraint<TAssertions> NotHaveMembers(params string[] memberNames)
+            => NotHaveMembers(memberNames, string.Empty);
+
+        public AndConstraint<TAssertions> NotHaveMembers(IEnumerable<string> memberNames, string because = "", params object[] reasonArgs) {
+            var actualNames = Subject.GetAllMembers(((ModuleScope)OwnerScope.GlobalScope).Module.InterpreterContext).Keys.ToArray();
```
Comment (AlexanderSher) [inline src/Analysis/Engine/Test/FluentAssertions/AnalysisValueAssertions.cs]: `((ModuleScope)OwnerScope.GlobalScope).Module.InterpreterContext` is used in several places. Let's extract it into helper method.

File: src/Analysis/Engine/Test/AstAnalysisTests.cs (hunk 1 above)
Comment (AlexanderSher) [reply src/Analysis/Engine/Test/AstAnalysisTests.cs]: Is it actually possible in VSC to change these parameters (Limits, SearchPaths) after server is started?

Comment (MikhailArkhipov) [reply src/Analysis/Engine/Test/AstAnalysisTests.cs]: Currently no. LS restart is required.

Comment (AlexanderSher) [reply src/Analysis/Engine/Test/AstAnalysisTests.cs]: In this case, maybe we should simplify these tests by setting all these values from the beginning? We can add another overload of `ServerExtensions.InitializeAsync`.

Comment (MikhailArkhipov) [reply src/Analysis/Engine/Test/AstAnalysisTests.cs]: But is actually good to set them later. LS doesn't do it for simplicity but it really should not be requiring restart. So at some point we should handle change.

Comment (AlexanderSher) [reply src/Analysis/Engine/Test/AstAnalysisTests.cs]: Ok

File: src/Analysis/Engine/Impl/PythonAnalyzer.cs [hunk 3]
```diff
@@ -622,7 +622,16 @@ class LazyModuleEnumerator {
 
         public AnalysisLimits Limits {
             get { return _limits; }
-            set { _limits = value; }
+            set {
+                var limits = _limits;
+                _limits = value;
+
+                if ((limits == null && _limits != null)
```
Comment (AlexanderSher) [inline src/Analysis/Engine/Impl/PythonAnalyzer.cs]: Can it be the other way: `limits != null && _limits == null`?

File: src/Analysis/Engine/Impl/Values/DictionaryInfo.cs [hunk 4]
```diff
@@ -208,7 +208,7 @@ public DictionaryInfo(IPythonProjectEntry declaringModule, Node node)
             }
         }
 
-        public override PythonMemberType MemberType => PythonMemberType.Field;
+        public override PythonMemberType MemberType => PythonMemberType.Instance;
```
Comment (AlexanderSher) [inline src/Analysis/Engine/Impl/Values/DictionaryInfo.cs]: Is it for some other bug?

File: src/Analysis/Engine/Impl/PythonAnalyzer.cs (hunk 3 above)
Comment (MikhailArkhipov) [reply src/Analysis/Engine/Impl/PythonAnalyzer.cs]: Actually I think it should be using default and never be null

**Pull request merged**